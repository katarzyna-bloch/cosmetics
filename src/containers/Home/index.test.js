import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Home } from '.'

jest.mock('react-slick', () => {
  const ProductsSlider = () => 'ProductsSlider'
  return ProductsSlider
})

const testProduct = { 
  brand: "Rimmel",
  categoryId: "3",
  description: "kremowa pomadka",
  hasDiscount: true,
  id: "22",
  img: "pomadka.jpg",
}

const testBanner = {
  id: '3',
  img: 'makeup.jpg',
  url: '/categories/makeup', 
  text: 'Różnorodne rabaty dla Ciebie. Zobacz wybrane produkty w super cenach!',
}

const props = {
  fetchSaleProducts: jest.fn(),
  fetchTopProducts: jest.fn(),
  fetchBanners: jest.fn(),
  saleProducts: [ testProduct ],
  topProducts: [ testProduct ],
  banners: [ testBanner ],
}

it('should get the data correctly', () => {
  const wrapped = mount(<Home {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()

  wrapped.update()
  expect(props.fetchSaleProducts).toHaveBeenCalled()
  expect(props.fetchTopProducts).toHaveBeenCalled()
  expect(props.fetchBanners).toHaveBeenCalled()
})
