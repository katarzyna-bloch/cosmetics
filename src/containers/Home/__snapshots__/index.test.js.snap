// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get the data correctly 1`] = `
<Home
  banners={
    Array [
      Object {
        "id": "3",
        "img": "makeup.jpg",
        "text": "Różnorodne rabaty dla Ciebie. Zobacz wybrane produkty w super cenach!",
        "url": "/categories/makeup",
      },
    ]
  }
  fetchBanners={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  fetchSaleProducts={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  fetchTopProducts={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  saleProducts={
    Array [
      Object {
        "brand": "Rimmel",
        "categoryId": "3",
        "description": "kremowa pomadka",
        "hasDiscount": true,
        "id": "22",
        "img": "pomadka.jpg",
      },
    ]
  }
  topProducts={
    Array [
      Object {
        "brand": "Rimmel",
        "categoryId": "3",
        "description": "kremowa pomadka",
        "hasDiscount": true,
        "id": "22",
        "img": "pomadka.jpg",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-iCfLBT dWFFgQ"
    >
      <Banners
        banners={
          Array [
            Object {
              "id": "3",
              "img": "makeup.jpg",
              "text": "Różnorodne rabaty dla Ciebie. Zobacz wybrane produkty w super cenach!",
              "url": "/categories/makeup",
            },
          ]
        }
      >
        <Carousel
          items={
            Array [
              <Banner
                banner={
                  Object {
                    "id": "3",
                    "img": "makeup.jpg",
                    "text": "Różnorodne rabaty dla Ciebie. Zobacz wybrane produkty w super cenach!",
                    "url": "/categories/makeup",
                  }
                }
              />,
            ]
          }
          settings={
            Object {
              "autoplay": true,
              "dots": true,
              "slidesToShow": 1,
            }
          }
        >
          <styled.div
            sideMargins={0}
            withBottomMargin={true}
          >
            <div
              className="sc-jRQAMF ftbaup"
            >
              <ProductsSlider
                autoplay={true}
                dots={true}
                draggable={false}
                gap={0}
                infinite={true}
                slidesToShow={1}
                speed={200}
              >
                ProductsSlider
              </ProductsSlider>
            </div>
          </styled.div>
        </Carousel>
      </Banners>
      <Styled(styled.h5)>
        <h5
          className="sc-bdvvaa sc-ksdxAp cRcTHD gIxfol"
        >
          W tym tygodniu w promocji
        </h5>
      </Styled(styled.h5)>
      <ProductsSlider
        products={
          Array [
            Object {
              "brand": "Rimmel",
              "categoryId": "3",
              "description": "kremowa pomadka",
              "hasDiscount": true,
              "id": "22",
              "img": "pomadka.jpg",
            },
          ]
        }
      >
        <Carousel
          items={
            Array [
              <Styled(ProductListItem)
                product={
                  Object {
                    "brand": "Rimmel",
                    "categoryId": "3",
                    "description": "kremowa pomadka",
                    "hasDiscount": true,
                    "id": "22",
                    "img": "pomadka.jpg",
                  }
                }
              />,
            ]
          }
          settings={
            Object {
              "gap": 30,
              "responsive": Array [
                Object {
                  "breakpoint": 768,
                  "settings": Object {
                    "infinite": true,
                    "slidesToScroll": 2,
                    "slidesToShow": 2,
                  },
                },
                Object {
                  "breakpoint": 414,
                  "settings": Object {
                    "slidesToScroll": 1,
                    "slidesToShow": 1,
                  },
                },
              ],
              "slidesToScroll": 2,
              "slidesToShow": 3,
            }
          }
        >
          <styled.div
            sideMargins={15}
            withBottomMargin={false}
          >
            <div
              className="sc-jRQAMF iQnpzi"
            >
              <ProductsSlider
                dots={false}
                draggable={false}
                gap={30}
                infinite={true}
                responsive={
                  Array [
                    Object {
                      "breakpoint": 768,
                      "settings": Object {
                        "infinite": true,
                        "slidesToScroll": 2,
                        "slidesToShow": 2,
                      },
                    },
                    Object {
                      "breakpoint": 414,
                      "settings": Object {
                        "slidesToScroll": 1,
                        "slidesToShow": 1,
                      },
                    },
                  ]
                }
                slidesToScroll={2}
                slidesToShow={3}
                speed={200}
              >
                ProductsSlider
              </ProductsSlider>
            </div>
          </styled.div>
        </Carousel>
      </ProductsSlider>
      <Styled(styled.h5)>
        <h5
          className="sc-bdvvaa sc-ksdxAp cRcTHD gIxfol"
        >
          Top oferty
        </h5>
      </Styled(styled.h5)>
      <ProductsSlider
        products={
          Array [
            Object {
              "brand": "Rimmel",
              "categoryId": "3",
              "description": "kremowa pomadka",
              "hasDiscount": true,
              "id": "22",
              "img": "pomadka.jpg",
            },
          ]
        }
      >
        <Carousel
          items={
            Array [
              <Styled(ProductListItem)
                product={
                  Object {
                    "brand": "Rimmel",
                    "categoryId": "3",
                    "description": "kremowa pomadka",
                    "hasDiscount": true,
                    "id": "22",
                    "img": "pomadka.jpg",
                  }
                }
              />,
            ]
          }
          settings={
            Object {
              "gap": 30,
              "responsive": Array [
                Object {
                  "breakpoint": 768,
                  "settings": Object {
                    "infinite": true,
                    "slidesToScroll": 2,
                    "slidesToShow": 2,
                  },
                },
                Object {
                  "breakpoint": 414,
                  "settings": Object {
                    "slidesToScroll": 1,
                    "slidesToShow": 1,
                  },
                },
              ],
              "slidesToScroll": 2,
              "slidesToShow": 3,
            }
          }
        >
          <styled.div
            sideMargins={15}
            withBottomMargin={false}
          >
            <div
              className="sc-jRQAMF iQnpzi"
            >
              <ProductsSlider
                dots={false}
                draggable={false}
                gap={30}
                infinite={true}
                responsive={
                  Array [
                    Object {
                      "breakpoint": 768,
                      "settings": Object {
                        "infinite": true,
                        "slidesToScroll": 2,
                        "slidesToShow": 2,
                      },
                    },
                    Object {
                      "breakpoint": 414,
                      "settings": Object {
                        "slidesToScroll": 1,
                        "slidesToShow": 1,
                      },
                    },
                  ]
                }
                slidesToScroll={2}
                slidesToShow={3}
                speed={200}
              >
                ProductsSlider
              </ProductsSlider>
            </div>
          </styled.div>
        </Carousel>
      </ProductsSlider>
    </div>
  </styled.div>
</Home>
`;
